### PR TITLE
Assign seat to trip and show it in expandable Seat Bids section

### PR DESCRIPTION
**Pull request by nowshin-108**
Seat assignment: 

1. Schema Update:
   - Introduced two main tables: one for trips related to a single user, and another to store all passengers for a specific flight.
   - Reason: This design separates user-specific trip data from flight-specific passenger data, allowing for more efficient querying and better scalability.
   - Advantage: Enables easier implementation of features like seat bidding

2. Backend Seat Assignment:
   - Added random seat generation and assignment to the backend.
   - Implemented an 'attempt' loop to handle potential seat conflicts during assignment.
   - Advantage: Prevents issues with concurrent bookings and provides a more robust seat assignment process.

3. Enhanced Trip Details Endpoint:
   - Updated '/trips/:tripId' route to include seat information.
   - Eliminates the need for a separate seat info endpoint.
   - Advantage: Reduces the number of API calls needed by the frontend, improving performance.

4. Seat Bids Section UI Enhancement: 
    - Transformed Seat Bids into an expandable section, matching the style of other manage sections 
    - Implemented React Icons (IoIosArrowUp and IoIosArrowDown) for expand/collapse indication 
    - Improved overall layout and styling of the Seat Bids section 
   
Screenshot 2024-07-18 at 3 54 45 PM

Screenshot 2024-07-18 at 3 54 55 PM

### CODE REVIEW

File: frontend/src/Components/HomePage/UpcomingTrips/TripDetails.jsx [hunk 1]
```diff
@@ -50,6 +52,12 @@ const getUserLocation = useCallback(() => {
     });
 }, []);
 
+// Generate Random seat
+const generateRandomSeat = () => {
```
Comment (alantulais) [inline frontend/src/Components/HomePage/UpcomingTrips/TripDetails.jsx:56]: For random seat, I'd prefer us to store it in the database when we create the trip. You also need to make sure you get unique random seat for the same flight.

**Review by alantulais — CHANGES_REQUESTED** [review]

File: frontend/src/Components/HomePage/UpcomingTrips/TripDetails.css [hunk 2]
```diff
@@ -227,6 +227,25 @@ p {
...
+    color: #333;
+}
+
+.seat-bids-content {
+    padding: 10px 0;
+    border-bottom: 1px solid #e0e0e0;
+}
+
+.seat-bids-content p {
+    margin: 0;
```
Comment (abeadam) [inline frontend/src/Components/HomePage/UpcomingTrips/TripDetails.css:246]: Would margin be zero by default, one recommendation is [BEM](https://getbem.com/naming/) to make css cleaner, don't worry about it here though.

File: frontend/src/Components/HomePage/UpcomingTrips/TripDetails.jsx [hunk 3]
```diff
@@ -306,6 +322,20 @@ return (
     <h2>Manage</h2>
     <div className="manage-section">
         <ul>
+        <li onClick={() => setIsSeatBidsExpanded(!isSeatBidsExpanded)}>
```
Comment (abeadam) [inline frontend/src/Components/HomePage/UpcomingTrips/TripDetails.jsx:325]: cleaner to normally not have functions in the middle of the jsx

File: frontend/src/Components/HomePage/UpcomingTrips/TripDetails.jsx (hunk 1 above)
Comment (nowshin-108) [reply frontend/src/Components/HomePage/UpcomingTrips/TripDetails.jsx]: Fixed it!

File: backend/routes/flightStatus.js [hunk 4]
```diff
@@ -49,43 +49,100 @@ router.get('/add-trip/flight-status', authenticateUser, async (req, res) => {
...
+                    await prisma.passenger.create({
+                        data: {
+                            userId,
+                            flightId: flight.flightId,
+                            seatNumber
+                        }
+                    });
+                    seatAssigned = true;
+                } catch (error) {
+                    if (error.code === 'P2002') {
```
Comment (alantulais) [inline backend/routes/flightStatus.js:106]: Ok so we are checking this at database level with unique restraint?

File: backend/routes/flightStatus.js [hunk 5]
```diff
@@ -49,43 +49,100 @@ router.get('/add-trip/flight-status', authenticateUser, async (req, res) => {
...
+                const seat = ['A', 'B', 'C', 'D', 'E', 'F'][Math.floor(Math.random() * 6)];
+                return `${row}${seat}`;
+            };
+
+            let seatAssigned = false;
+            let attempts = 0;
+            let seatNumber;
+            
+            // Allow system to try assigning a random seat multiple times in case the number is already taken
+            while (!seatAssigned && attempts < 100) {
```
Comment (alantulais) [inline backend/routes/flightStatus.js:94]: LGTM!

**Review by alantulais — APPROVED** [review]

**Pull request closed without merging**